yt/data_objects/data_containers.py

```diff
@@ -808,11 +815,64 @@ def create_firefly_object(
             )
             datadir = JSONdir
 
+        if match_any_particle_types is None:
+            # not specified, switching to (temporary) default
+            issue_deprecation_warning(
+                "match_any_particle_types wasn't specified. Its current default "
+                "value (True) will be changed to False in the future. "
+                "Pass an actual value (True or False) to silence this warning. ",
+                since="4.2",
+            )
+            # Note that all of the tests in test_firefly.py have this explicitly
+            # specified (and labeled) to avoid this warning. They will need to be
+            # changed when this behavior changes in the future
+            match_any_particle_types = True
+
         ## initialize a firefly reader instance
         reader = firefly.data_reader.Reader(
             datadir=datadir, clean_datadir=True, **kwargs
         )
 
+        ## Ensure at least one field type contains every field requested
+        if match_any_particle_types:
+            # Need to keep previous behavior: single string field names that
+            # are ambiguous should bring in any matching ParticleGroups instead
+            # of raising an error
+            # This can be expanded/changed in the future to include field
+            # tuples containing some sort of special "any" ParticleGroup
+            unambiguous_fields_to_include = []
+            unambiguous_fields_units = []
+            for field, field_unit in zip(fields_to_include, fields_units):
+                if isinstance(field, tuple):
+                    # skip tuples, they'll be checked with _determine_fields
+                    unambiguous_fields_to_include.append(field)
+                    unambiguous_fields_units.append(field_unit)
+                    continue
+                _, candidates = self.ds._get_field_info_helper(field)
+                if len(candidates) < 2:
+                    # Field is unambiguous, add in tuple form
+                    # This should be equivalent to _tupleize_field
+                    unambiguous_fields_to_include.append(candidates[0])
+                    unambiguous_fields_units.append(field_unit)
```

Comment:
0 candidates shouldn't be a valid case, and since we're using `candidates[0]` it's clear we're already assuming we have exactly one here
```suggestion
                if len(candidates) == 1:
                    # Field is unambiguous, add in tuple form
                    # This should be equivalent to _tupleize_field
                    unambiguous_fields_to_include.append(candidates[0])
                    unambiguous_fields_units.append(field_unit)
```